From a9c40a16bd5f5a2a7f5efa1c5f6d50bfba9344f4 Mon Sep 17 00:00:00 2001
From: Jared Morrow <jm@joyent.com>
Date: Thu, 11 Jan 2018 21:14:38 +0000
Subject: [PATCH] MANTA-3553 Muppet should recover when Zookeeper returns after
 being down Reviewed by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>
 Approved by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>

---
 muppet.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/muppet.js b/muppet.js
index 6ab8d19..f7f8290 100644
--- a/muppet.js
+++ b/muppet.js
@@ -284,10 +284,11 @@ function zookeeper(cfg) {
     /*
      * zkstream attempts to retry the connection to zookeeper. On retry
      * exhaustion it will emit a `failed` event.
+     * For this we only log an error, because when ZK comes back up, zkstream
+     * will automatically reconnect and re-arm the watchers (if they existed)
      */
     zk_client.on('failed', function onFailed(err) {
-        cfg.log.fatal(err, 'ZooKeeper: received failed event');
-        process.exit(1);
+        cfg.log.error(err, 'ZooKeeper: received failed event');
     });
 }
 
-- 
2.21.0

